packages/perseus-core/src/utils/extract-perseus-ai-data.test.ts

```diff
@@ -0,0 +1,1438 @@
+                type: "image",
+                options: {
+                    alt: "image alt text",
+                    backgroundImage: {
+                        url: "",
+                        width: 100,
+                        height: 100,
+                    },
+                },
+            },
+        } as const;
+        const content = injectWidgets(
+            "Content with an image [[☃ image 1]]",
+            widgets,
+        );
+        expect(content).toEqual(
+            'Content with an image <img id="image 1" alt="image alt text">',
+        );
+    });
+
+    it("should inject label-image widget into the content", () => {
+        const widgets: PerseusWidgetsMap = {
+            "label-image 1": {
+                type: "label-image",
+                options: {
+                    choices: ["answer 1", "answer 2"],
+                    markers: [
+                        {
+                            label: "Label 1",
+                            answers: ["answer 1"],
+                            x: 10,
+                            y: 10,
+                        },
+                        {
+                            label: "Label 2",
+                            answers: ["answer 2"],
+                            x: 20,
+                            y: 10,
+                        },
+                    ],
+                    imageAlt: "Alt text for the image",
+                    imageUrl: "",
+                    imageHeight: 100,
+                    imageWidth: 100,
+                    hideChoicesFromInstructions: false,
+                    multipleAnswers: false,
+                    static: false,
+                },
+            },
+        } as const;
+        const content = injectWidgets("[[☃ label-image 1]]", widgets);
+        expect(content).toEqual(
+            "[An image with dots that user needs to label. Label choices: [answer 1, answer 2]. Image alt text: Alt text for the image]",
+        );
+    });
+
+    it("should inject radio widget into the content", () => {
+        const widgets: PerseusWidgetsMap = {
+            "radio 1": {
+                type: "radio",
+                options: {
+                    choices: [
+                        {
+                            id: "0-0-0-0-0",
+                            content: "choice 1",
+                            correct: true,
+                        },
+                        {
+                            id: "1-1-1-1-1",
+                            content: "choice 2",
+                            correct: false,
+                        },
+                    ],
+                },
+            },
+        };
+        const content = injectWidgets(
+            "Content with a radio\n[[☃ radio 1]]",
+            widgets,
+        );
+        expect(content).toEqual("Content with a radio\nchoice 1\nchoice 2");
+    });
+
+    it("should inject radio widget into the content with randomization note", () => {
```

Comment:
Maybe these additional tests are overkill, but it seemed nice to have some confidence that each widget is handled correctly, so that we can quickly catch any regressions. 

We could possibly solve this randomization note eventually, but it would require us to pass in the shuffle seed data and translated strings - so we'll look into that at some point in the future.